better_budget.py

```diff
@@ -36,19 +39,27 @@ def compare_budget(dataframes, budget, total):
 			print('Oops! In your {} budget, you have spent ${:.2f}, which is ${:.2f} over budget.'.format(key, spent, abs(saved)))
 	print('there is {} unaccounted for.'.format(total-total_spent))
 
-def plot_budget(dataframes, budget_dict):
+def plot_budget(dataframes, budget_dict, keyword_dict):
 	cat_totals = {k: abs(v['Amount'].sum()) for k, v in dataframes.items()}
 	cat_labels = [str(k) for k in dataframes]
 
 	fig, ax = plt.subplots()
+        # plot budgeted amounts
 	plt.bar(cat_labels, [budget_dict[l] for l in cat_labels], alpha=0.6)
-	plt.bar(cat_labels, [cat_totals[l] for l in cat_labels], alpha=0.6)
+        # plot spent amounts, broken down first by category then by keyword
+        colors = ['C{}'.format(i) for i in range(1, 10)]
+        for cat in keyword_dict:
+            kw_amounts = amounts_by_keyword(dataframes[cat], keyword_dict[cat])
+            sorted_keys = sorted(kw_amounts, key=lambda x: kw_amounts[x], reverse=True)
+            for i, k in enumerate(sorted_keys):
+                bottom = sum(sorted(kw_amounts.values(), reverse=True)[:i])
+                plt.bar(cat, kw_amounts[k], bottom=bottom, color=colors[i], alpha=0.6)
```

Comment:
Hmm initially I thought this graph was needing a legend, but these really should just be a separate graph.

You're breaking down each keyword within each column, but every keyword is unique. If thats the case then it's misleading to have multiple blue columns when every blue column is a different value. What's the point of having them share a color?

Consider adding a `--details` flag that will print out a graph for each category that would benefit from this functionality. (this way we don't have a hundred graphs that are 100% one thing)

Reply:
I'm honestly not sure how flags work :/

Reply:
How they work or how to implement them?

Reply:
Okay, now that I'm with Daniel and talking about your comments etc, I just want to understand your suggestion:

- Do you want each category to be a separate graph, rather than side-by-side?
- I think if I'm reading your comments correctly, you don't like that the colors are re-used and therefore misleading.

The functionality of the graph, as I wrote it, is to see side-by-side what I'm spending the most money on. I _don't_ want individual graphs, since then I'd have to rapidly switch between them all to make comparisons, and that could get messy with resized y axes. Rather, I'd like to see them all in a line. I _do_ agree that the same colors could be misleading, but I'm moreso using the colors to see by eye if I'm spending a good chunk of my "fun" money in the same place, or my "travel" money in the same place, etc.
Does this make sense?

Reply:
> I think if I'm reading your comments correctly, you don't like that the colors are re-used and therefore misleading.

Yes, if a graph has multiple colors I think most people would expect each color to have a meaning.

Stacked bar graphs are also hard to read since there is no standard with them. Sometimes people have the base at the x axis sometimes the base is at the top of the previous bar.

I don't see what your reason is to avoid multiple graphs, to avoid scrolling or because it makes previous months harder to compare to? My idea was more to keep the existing graph then post subsequent graphs with a breakdown on each category. That way you have a graph for total comparisons, and graphs for comparisons within each category. Since these are percentage comparisons, my instinct for the individual categories is to use pie charts with a legend.

Or a print out to the right of the graph with percentage breakdowns, something like
```
Food: (30% tech), (70% fresh)
```
or
```
Food:
    Tech: 30%
    Fresh: 70%
```

Just something so you can reap the benefits of seeing the breakdowns. As of right now it just shows *something* is more than *something else*, which is really limiting.